lib/proc.c

```diff
@@ -477,6 +477,51 @@ grn_proc_options_parsev(grn_ctx *ctx,
                                           args);
 }
 
+static void
+grn_proc_func_generate_cache_key_add(grn_ctx *ctx,
+                                     grn_obj *cache_key,
+                                     grn_obj *arg)
+{
+  switch (arg->header.type) {
+  case GRN_TABLE_HASH_KEY :
+    GRN_HASH_EACH_BEGIN(ctx, (grn_hash *)arg, cursor, id) {
+      void *key;
+      unsigned int key_size;
+      void *value;
+      int value_size = grn_hash_cursor_get_key_value(ctx,
+                                                     cursor,
+                                                     &key,
+                                                     &key_size,
+                                                     &value);
+      GRN_TEXT_PUT(ctx, cache_key, key, key_size);
+      GRN_TEXT_PUTC(ctx, cache_key, '\1');
+      if (value_size == sizeof(grn_obj *)) {
```

Comment:
I'm not confident but isn't it `grn_obj` , not `grn_obj *` ?

Reply:
Good catch! You're right!